cdk/lib/mobile-save-for-later.ts

```diff
@@ -1,19 +1,134 @@
       templateFile: yamlTemplateFilePath,
     });
+
+    const app = "mobile-save-for-later";
+
+    const commonLambdaProps = {
+      runtime: Runtime.JAVA_8, // We should upgrade to Java 11 in a future PR
+      app,
+      fileName: `${app}.jar`,
+      // N.B. these lambdas previously used very specific, low memory sizes (384 & 394); odd choices for a Scala Lambda
+    };
+
+    const commonEnvironmentVariables = {
+      App: app,
+      Stack: this.stack,
+      Stage: this.stage,
+      IdentityApiHost: props.identityApiHost,
+    };
+
+    const saveArticlesLambda = new GuLambdaFunction(
+      this,
+      "save-articles-lambda",
+      {
+        handler: "com.gu.sfl.lambda.SaveArticlesLambda::handleRequest",
+        functionName: `mobile-save-for-later-SAVE-cdk-${this.stage}`,
+        environment: {
+          ...commonEnvironmentVariables,
+          SavedArticleLimit: "1000",
+        },
+        ...commonLambdaProps,
+      }
+    );
+
+    const fetchArticlesLambda = new GuLambdaFunction(
+      this,
+      "fetch-articles-lambda",
+      {
+        handler: "com.gu.sfl.lambda.FetchArticlesLambda::handleRequest",
+        functionName: `mobile-save-for-later-FETCH-cdk-${this.stage}`,
+        timeout: Duration.seconds(20),
+        reservedConcurrentExecutions: this.stage === "PROD" ? 50 : 1,
+        environment: commonEnvironmentVariables,
+        ...commonLambdaProps,
+      }
+    );
+
+    [saveArticlesLambda, fetchArticlesLambda].map((lambda) => {
+      // permissions for the departmental standard are provided by cdk 'for free' but we cannot
+      // use them here because our parameter path used differs from the departmental standard
+      lambda.addToRolePolicy(
+        new PolicyStatement({
+          actions: ["ssm:GetParametersByPath"],
+          resources: [
+            `arn:aws:ssm:${this.region}:${this.account}:parameter/${app}/${this.stage}/${this.stack}`,
+          ],
+        })
+      );
+      lambda.addToRolePolicy(
+        new PolicyStatement({
+          actions: ["cloudwatch:putMetricData"],
+          resources: ["*"],
+        })
+      );
+      lambda.addToRolePolicy(
+        new PolicyStatement({
+          actions: [
+            "dynamodb:GetItem",
+            "dynamodb:PutItem",
+            "dynamodb:UpdateItem",
+            "dynamodb:Query",
+          ],
+          resources: [
+            yamlDefinedResources
+              .getResource("SaveForLaterDynamoTable")
+              // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#aws-resource-dynamodb-table-return-values
+              .getAtt("Arn")
+              .toString(),
+          ],
+        })
+      );
+    });
+
+    const saveForLaterApi = new GuApiGatewayWithLambdaByPath(this, {
+      app,
+      restApiName: `${app}-api-${this.stage}`,
+      monitoringConfiguration: { noMonitoring: true }, //TODO: configure monitoring (this is not setup in current CFN)
```

Comment:
So I think we need to add `SaveForLaterReadThrottleEvents` and `SaveForLaterWriteThrottleEvents` cloudwatch events in subsequent PRs?

Reply:
You're right; for now both of those alarms are still configured via the YAML template. Since the alarms relate to a Dynamo table which is used by old (CFN-defined) and new (cdk-defined) versions of the lambdas, these will continue to serve their purpose even though we have not defined them using TypeScript.

However this `monitoringConfiguration` prop will actually help us to configure an alarm for 5XX responses (which is a superset of errors, where one subset is errors caused by read/write throttle events). Personally I would opt for a 5XX alarm and drop these other two more precise alarms. I will do that via a follow-up PR after the migration as I think it warrants its own discussion.

Reply:
Got it.  Thank you Jacob.